example/Makefile

```diff
@@ -3,7 +3,7 @@ CXX := c++
 endif
 
 ifndef CXXFLAGS
-CXXFLAGS := -std=c++11 -Wall -I .. -o
+CXXFLAGS := -std=c++11 -Wall -Wextra -Wshadow -Weffc++ -Wno-unused-parameter -Wno-unused-local-typedefs -Wno-long-long -Wstrict-aliasing -Werror -pedantic -std=c++1y -Wformat=2 -Wmissing-include-dirs -Wsync-nand -Wuninitialized -Wconditionally-supported -Wconversion -Wuseless-cast -Wzero-as-null-pointer-constant -I .. -o
```

Comment:
I will use these privately to improve `enum.h` and the tests and examples, but I prefer to leave the flags out of the `Makefile`s for now. The main reason is that warning flags are compiler-specific. For example, the ones here are only compatible with g++ 4.9 and 5.1. Only g++ 4.9 was strict enough about shadowing to complain about some of my names differing by underscores :) My default system compiler (clang++ 3.6) refuses to run with these flags, for example.

The topic branch I mentioned has a Python script that attempts to build everything on a bunch of different compilers. I will add flag awareness to it instead of here. In the meantime, it's better if someone who downloads the project is able to build it with their system's recent GCC-like compiler with the default settings in the `Makefile`s.